Content.Server/Automod/AutomodManager.cs

```diff
@@ -0,0 +1,260 @@
+﻿using System.Diagnostics;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Content.Server.Database;
+using Content.Shared.Automod;
+using Content.Shared.Database;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Automod;
+
+public sealed class AutomodManager : IAutomodManager, IPostInjectInit
+{
+    [Dependency] private readonly IPrototypeManager _protoMan = default!;
+    [Dependency] private readonly IEntityManager _entMan = default!;
+    [Dependency] private readonly ILogManager _logMan = default!;
+    [Dependency] private readonly IServerDbManager _db = default!;
+
+    private ISawmill _log = default!;
+
+    // Filters
+    private readonly Dictionary<AutomodTarget, Dictionary<Regex, AutomodFilterDef>> _regexFilters = new();
+
+    #region Initialize
+
+    public void Initialize()
+    {
+        IoCManager.InjectDependencies(this);
+
+        LoadAutomodFiltersFromDatabase();
+    }
+
+    private async void LoadAutomodFiltersFromDatabase()
+    {
+        var filters = await _db.GetAllAutomodFiltersAsync();
+        foreach (var filter in filters)
+        {
+            AddFilter(filter);
+        }
+    }
+
+    void IPostInjectInit.PostInject()
+    {
+        _log = _logMan.GetSawmill("automod");
+    }
+
+    #endregion
+
+    #region Filter
+
+    public bool Filter(AutomodTarget target, string inputText, ICommonSession session)
+    {
+        return RegexFilter(target, inputText, session);
+    }
+
+    private bool RegexFilter(AutomodTarget target, string inputText, ICommonSession session)
+    {
+        // Ensure that only 1 bit is set
+        Debug.Assert(target != 0 && (target & (target - 1)) == 0);
+
+        var blocked = true;
```

Comment:
Uh shouldn't this be called `passes`?

Reply:
Yes, thanks. [7b37468](https://github.com/space-wizards/space-station-14/pull/28570/commits/7b374688361859ed9be1135a22d7ad83b11d7cb0)